libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -87,6 +87,27 @@ void AP_Camera::take_picture()
     primary->take_picture();
 }
 
+void AP_Camera::take_picture(uint8_t instance)
+{
+    WITH_SEMAPHORE(_rsem);
+
+    // if instance is 0 take picture for all existing instances
+    // else take picture only for given instance
+    if (instance == 0) {
```

Comment:
So this interprets the interpret the instance field differently in this method compared to other methods? Imagine the confusion to callers and the number of bugs this will lead to.

Reply:
@rmackay9 Other methods will also going to use the same in future. But for now could you suggest any other approach.

Reply:
Hi @khanasif786,

I think the best thing to do would be to change the method with no argument so that it affects all cameras.  Then we can simplify the method which does take an instance to only affect the specified instance.